### PR TITLE
Update aws_apigatewayv2_integration documentation for private integration

### PR DESCRIPTION
**Pull request by BertrandMarechal**
- Adds use case on `integration_type` for HTTP_PROXY
- Adds an example for a pirvate integration with a load balancer listener

### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

### CODE REVIEW

File: website/docs/r/apigatewayv2_integration.html.markdown [hunk 1]
```diff
@@ -63,13 +63,37 @@ resource "aws_apigatewayv2_integration" "example" {
...
+  integration_type    = "HTTP_PROXY"
+  integration_uri     = aws_lb_listener.example.arn
+  
+  connection_type = "VPC_LINK"
+  connection_id   = aws_apigatewayv2_vpc_link.example.id
+
+  request_parameters = {
+    "QueueUrl"    = "$request.header.queueUrl"
+    "MessageBody" = "$request.body.message"
+  }
```
Comment (ewbankkit) [inline website/docs/r/apigatewayv2_integration.html.markdown:82]: I think it's better to remove the `request_parameters` block (particularly with these values) as it is usually applicable to an AWS service integration.

**Review by ewbankkit — CHANGES_REQUESTED** [review]
@BertrandMarechal Thanks for the contribution.
LGTM apart from the suggested change to remove `request_parameters`.

File: website/docs/r/apigatewayv2_integration.html.markdown [hunk 2]
```diff
@@ -63,13 +63,37 @@ resource "aws_apigatewayv2_integration" "example" {
...
 
+### Private Integration
+
+```hcl
+resource "aws_apigatewayv2_integration" "example" {
+  api_id              = aws_apigatewayv2_api.example.id
+  credentials_arn     = aws_iam_role.example.arn
+  description         = "Example with a load balancer"
+  integration_type    = "HTTP_PROXY"
+  integration_uri     = aws_lb_listener.example.arn
```
Comment (ewbankkit) [inline website/docs/r/apigatewayv2_integration.html.markdown:74]: Align to please linter:

```suggestion
  api_id           = aws_apigatewayv2_api.example.id
  credentials_arn  = aws_iam_role.example.arn
  description      = "Example with a load balancer"
  integration_type = "HTTP_PROXY"
  integration_uri  = aws_lb_listener.example.arn
```

File: website/docs/r/apigatewayv2_integration.html.markdown [hunk 3]
```diff
@@ -63,13 +63,37 @@ resource "aws_apigatewayv2_integration" "example" {
...
+  
+  connection_type = "VPC_LINK"
+  connection_id   = aws_apigatewayv2_vpc_link.example.id
+
+  request_parameters = {
+    "QueueUrl"    = "$request.header.queueUrl"
+    "MessageBody" = "$request.body.message"
+  }
+
```
Comment (ewbankkit) [inline website/docs/r/apigatewayv2_integration.html.markdown:83]: Remove whitespace.

```suggestion

```

File: website/docs/r/apigatewayv2_integration.html.markdown (hunk 1 above)
Comment (BertrandMarechal) [reply website/docs/r/apigatewayv2_integration.html.markdown]: Thanks, copy paste issue 👍

**Review by ewbankkit — APPROVED** [review]
LGTM.

**Comment by breathingdust** [comment]
Thanks @BertrandMarechal! 🚀

**Pull request merged**